Adding decomposition MPSPrep (#6896)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

* The requested `diff_method` is now validated when program capture is enabled.
  [(#6852)](https://github.com/PennyLaneAI/pennylane/pull/6852)

* The template `MPSPrep` now has a gate decomposition. This enables its use with any device.
  [(#6896)](https://github.com/PennyLaneAI/pennylane/pull/6896)

* The `qml.clifford_t_decomposition` has been improved to use less gates when decomposing `qml.PhaseShift`.
  [(#6842)](https://github.com/PennyLaneAI/pennylane/pull/6842)

* `null.qubit` can now execute jaxpr.
  [(#6924)](https://github.com/PennyLaneAI/pennylane/pull/6924)

> * Autograph can now be used with custom operations defined outside of the pennylane namespace.
>   [(#6931)](https://github.com/PennyLaneAI/pennylane/pull/6931)
> 

Author: ```suggestion
```

* The requested `diff_method` is now validated when program capture is enabled.
  [(#6852)](https://github.com/PennyLaneAI/pennylane/pull/6852)

> 

Reviewer: ```suggestion
```

* The requested `diff_method` is now validated when program capture is enabled.
  [(#6852)](https://github.com/PennyLaneAI/pennylane/pull/6852)

* The template `MPSPrep` now has a gate decomposition. This enables its use with any device.
  [(#6896)](https://github.com/PennyLaneAI/pennylane/pull/6896)

> 

Reviewer 2: Should changelog mention `right_canonicalize_mps()`?

* The requested `diff_method` is now validated when program capture is enabled.
  [(#6852)](https://github.com/PennyLaneAI/pennylane/pull/6852)

* The template `MPSPrep` now has a gate decomposition. This enables its use with any device.

>   `right_canonicalize_mps` functionality has also been added to transform an MPS into its right-canonical form.

Reviewer 3: ```suggestion
  The `right_canonicalize_mps` function has also been added to transform an MPS into its right-canonical form.
```
